wheatley/bot.py

```diff
@@ -157,7 +174,23 @@ def _on_size_change(self) -> None:
         self._check_number_of_bells()
         self._opening_row = generate_starting_row(self.number_of_bells, self.row_generator.custom_start_row)
         self._rounds = rounds(self.number_of_bells)
-        self._check_starting_row()
+        self._check_starting_row()  # Check that the current row gen is OK (otherwise warn the user)
+
+        # Make sure we have exclusive access to `self.next_row_generator`
+        with self.next_row_generator_lock:
+            if self.next_row_generator is not None and not self._check_number_of_bells(
+                self.next_row_generator, silent=True
+            ):
+                self.logger.warning("Next row gen needed too many bells, so is being removed.")
+                # If the `next_row_generator` can't be rung on the new stage, then remove it.
+                #
+                # If this isn't done, there would be a bug where Wheatley sees the tower size get
+                # smaller, but only checks `self.row_generator` for the new stage.  Meanwhile,
+                # `self.next_row_generator` has already been set (and therefore has the old stage) but,
+                # when `Look To` is called after the size change, is swapped into `self.row_generator`
+                # without checking its validity, thus causing Wheatley to crash
```

Comment:
My general feeling is this level of detail belongs in the commit message rather than code comments.
It's explaining a current bug and what other bits of code are currently doing, which can easily becomes out of date with future refactoring / features.

Reply:
Right I've done some cheeky rebasing, and moved it to a commit message